### ISSUE TITLE
NLP with mapped interpolant fails: hessian n_reverse() fails

### ISSUE THREAD

**Issue by jgillis** [casadi/casadi#1977]
```python
from casadi import *

np.random.seed(0)
    
d_knots = [list(np.linspace(0,1,5))]

data = np.random.random([len(e) for e in d_knots])
r = np.array(d_knots)

xyz = np.vstack(e.ravel(order='F') for e in r).ravel(order='F')

d_flat = data.ravel(order='F')

LUT = casadi.interpolant('name','linear',d_knots,d_flat)


H = LUT.map(10)


x = MX.sym("x",10)

gradient(sum2(H(x)),x)
```

The assertion "get_n_reverse()>0" on line 1534 of file "/home/jgillis/programs/casadi/casadi/core/function/function_internal.cpp" failed.

**Comment by jgillis** [comment]
Interpolant defines `fullJacobian` only; `LUT.reverse()` fails.
`LUT.jacobian_old` works since it checks `adjViaJac`.

`adjViaJac` logic fails here because `Map` defines `int get_n_reverse() const override { return 64;}` as opposed to `int get_n_forward() const override { return f_->get_n_reverse();}`, and omits fullJacobian.

**Comment by jgillis** [comment]
Seems to be fixed in 3.3
